frontend/src/lib/components/Alerts/views/EditAlert.tsx

```diff
@@ -0,0 +1,130 @@
+import { LemonInput } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { Form } from 'kea-forms'
+import { IconChevronLeft } from 'lib/lemon-ui/icons'
+import { LemonButton } from 'lib/lemon-ui/LemonButton'
+import { LemonField } from 'lib/lemon-ui/LemonField'
+import { LemonModal } from 'lib/lemon-ui/LemonModal'
+
+import { InsightShortId } from '~/types'
+
+import { alertLogic } from '../alertLogic'
+import { alertsLogic } from '../alertsLogic'
+
+interface EditAlertProps {
+    id: number | 'new'
+    insightShortId: InsightShortId
+    onCancel: () => void
+    onDelete: () => void
+}
+
+export function EditAlert({ id, insightShortId, onCancel, onDelete }: EditAlertProps): JSX.Element {
+    const logicProps = {
+        id,
+        insightShortId,
+    }
+    const logic = alertLogic(logicProps)
+    const alertslogic = alertsLogic({
+        id,
+        insightShortId,
+    })
+
+    const { alert, isAlertSubmitting, alertChanged } = useValues(logic)
+    const { deleteAlert } = useActions(alertslogic)
+
+    const _onDelete = (): void => {
+        if (id !== 'new') {
+            deleteAlert(id)
+            onDelete()
+        }
+    }
+
+    return (
+        <Form logic={alertLogic} props={logicProps} formKey="alert" enableFormOnSubmit className="LemonModal__layout">
+            <LemonModal.Header>
+                <div className="flex items-center gap-2">
+                    <LemonButton icon={<IconChevronLeft />} onClick={onCancel} size="xsmall" />
+
+                    <h3>{id === 'new' ? 'New' : 'Edit '} Alert</h3>
+                </div>
+            </LemonModal.Header>
+
+            <LemonModal.Content className="space-y-2">
+                {!alert ? (
+                    <div className="p-4 text-center">
+                        <h2>Not found</h2>
+                        <p>This alert could not be found. It may have been deleted.</p>
+                    </div>
+                ) : (
+                    <>
+                        <LemonField name="name" label="Name">
+                            <LemonInput placeholder="e.g. High error rate" />
+                        </LemonField>
+
+                        <LemonField
+                            name="target_value"
+                            label="Who do you want to notify about anomalies"
+                            help="Enter comma separated email addresses of the users you want notify about anomalies"
+                        >
+                            <LemonInput data-attr="alert-emails" placeholder="Enter an email address" />
+                        </LemonField>
+                        <LemonField name="anomaly_condition" label="Anomaly condition">
```

Comment:
Please let me know if there's a more elegant way to assign nested values of an object. I tried nesting LemonField-s one inside another, that didn't work

Reply:
You can use the [Group](https://keajs.org/docs/plugins/forms/#react-components) tag to nest fields, or just use an array as the field name. `name={['anomaly_condition', 'absolute_threshold']}`